virtcontainers/agent.go

```diff
@@ -146,27 +146,27 @@ type agent interface {
 	exec(sandbox *Sandbox, c Container, cmd Cmd) (*Process, error)
 
 	// startSandbox will tell the agent to start all containers related to the Sandbox.
-	startSandbox(sandbox Sandbox) error
+	startSandbox(sandbox *Sandbox) error
 
 	// stopSandbox will tell the agent to stop all containers related to the Sandbox.
-	stopSandbox(sandbox Sandbox) error
+	stopSandbox(sandbox *Sandbox) error
 
 	// createContainer will tell the agent to create a container related to a Sandbox.
 	createContainer(sandbox *Sandbox, c *Container) (*Process, error)
 
 	// startContainer will tell the agent to start a container related to a Sandbox.
-	startContainer(sandbox Sandbox, c *Container) error
+	startContainer(sandbox *Sandbox, c *Container) error
 
 	// stopContainer will tell the agent to stop a container related to a Sandbox.
-	stopContainer(sandbox Sandbox, c Container) error
```

Comment:
The rationale behind using pointers versus copies was to make sure the sandbox could not get modified by some functions that obviously didn't need it.
Now I understand your concern about memory being overused because we're generating a lot of copies and I think I pretty okay with changing things into pointers. But please submit this into a separate PR as it is an important change and this way we can discuss properly about it. I don't want this to be hidden through the monitor PR.

Reply:
Without it, I got CI errors about copying mutex.

Reply:
Ok, but please submit it as a separate PR.

Reply:
Then this PR cannot pass CI.

Reply:
That's fine, we'll merge the new PR, so that you can rebase this one.

Reply:
ok, here it is #263